File: app/src/test/java/com/mgaetan89/showsrage/fragment/LogsFragment_GetLogLevelForMenuId.java
package com.mgaetan89.showsrage.fragment;

import com.mgaetan89.showsrage.R;
import com.mgaetan89.showsrage.model.LogLevel;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class LogsFragment_GetLogLevelForMenuId {
	@Parameterized.Parameter(1)
	public LogLevel logLevel;

	@Parameterized.Parameter(0)
	public int menuId;

	@Test
	public void getLogLevelForMenuId() {
		assertThat(LogsFragment.Companion.getLogLevelForMenuId(this.menuId)).isEqualTo(this.logLevel);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{0, null},
				{R.id.menu_debug, LogLevel.DEBUG},
				{R.id.menu_error, LogLevel.ERROR},
				{R.id.menu_info, LogLevel.INFO},
				{R.id.menu_warning, LogLevel.WARNING},
				{R.id.menu_change_quality, null},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/LogsFragment_GetLogLevelForMenuId.kt
package com.mgaetan89.showsrage.fragment

import com.mgaetan89.showsrage.R
import com.mgaetan89.showsrage.model.LogLevel
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class LogsFragment_GetLogLevelForMenuId {
    @Parameterized.Parameter(1)
    var logLevel: LogLevel? = null

    @Parameterized.Parameter(0)
    var menuId: Int = 0

    @Test
    fun getLogLevelForMenuId() {
        assertThat(LogsFragment.getLogLevelForMenuId(this.menuId)).isEqualTo(this.logLevel)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any?>> {
            return listOf(
                    arrayOf<Any?>(0, null),
                    arrayOf<Any?>(R.id.menu_debug, LogLevel.DEBUG),
                    arrayOf<Any?>(R.id.menu_error, LogLevel.ERROR),
                    arrayOf<Any?>(R.id.menu_info, LogLevel.INFO),
                    arrayOf<Any?>(R.id.menu_warning, LogLevel.WARNING),
                    arrayOf<Any?>(R.id.menu_change_quality, null)
            )
        }
    }
}
